source/distributed/solution_transfer.cc

```diff
@@ -216,18 +216,18 @@ namespace parallel
                   const typename Triangulation<dim,dim>::CellStatus /*status*/,
                   void *data)
     {
-      double *data_store = reinterpret_cast<double *>(data);
+      typename VECTOR::value_type *data_store = reinterpret_cast<typename VECTOR::value_type *>(data);
 
       typename DH::cell_iterator cell(*cell_, dof_handler);
 
       const unsigned int dofs_per_cell=cell->get_fe().dofs_per_cell;
-      ::dealii::Vector<double> dofvalues(dofs_per_cell);
+      ::dealii::Vector<typename VECTOR::value_type> dofvalues(dofs_per_cell);
       for (typename std::vector<const VECTOR *>::iterator it=input_vectors.begin();
            it !=input_vectors.end();
            ++it)
         {
           cell->get_interpolated_dof_values(*(*it), dofvalues);
-          std::memcpy(data_store, &dofvalues(0), sizeof(double)*dofs_per_cell);
+          std::memcpy(data_store, &dofvalues(0), sizeof(typename VECTOR::value_type)*dofs_per_cell);
```

Comment:
Right. That's why I suggested the assertion, so that someone else can deal with it later ;-)